src/index.js

```diff
@@ -1,47 +1,18 @@
 const fullCorvidTypes = require("../fullCorvidTypes.json");
 const wixModulesNames = require("../wixModules.json");
 const { TS_CONFIG_PATHS } = require("./constants");
-const { getWidgetTypeDeclarations } = require("./widget");
-const getPageElementsTypeDeclarations = elementsMap =>
-  "type PageElementsMap = {\n" +
-  Object.keys(elementsMap)
-    .map(nickname => `  "#${nickname}": ${elementsMap[nickname]};\n`)
-    .join("") +
-  "}";
-
-const getModuleTypeDeclaration = name =>
-  `
-    declare module '${name}';
-    declare module '${name}/*';
-  `;
-
-const getDynamicTypes = ({ elementsMap, widgets }) => {
-  const dynamicTypes = [];
-  if (elementsMap) {
-    dynamicTypes.push({
-      path: "/elementsMap.d.ts",
-      content: getPageElementsTypeDeclarations(elementsMap)
-    });
-  }
-  if (widgets && widgets.length) {
-    widgets.forEach(({ name, events, members }) => {
-      dynamicTypes.push({
-        path: `/widgets/${name}.d.ts`,
-        content: getWidgetTypeDeclarations({ name, events, members })
-      });
-    });
-  }
-  return dynamicTypes;
-};
-
-const getNpmDependenciesTypesDeclarations = (dependencies = {}) => {
-  const dependenciesNames = Object.keys(dependencies);
-  if (!dependenciesNames.length) return [];
-  return dependenciesNames.map(packageName => ({
-    path: `/dependencies/${packageName}.${dependencies[packageName]}.d.ts`,
-    content: getModuleTypeDeclaration(packageName)
-  }));
-};
+const {
+  getAmbientModuleDeclaration,
+  getNpmDependenciesTypesDeclarations
+} = require("./dynamicTypes/npmDependencies");
+const {
+  getPageElementsTypeDeclarations,
+  getPageElementsRawDeclarations
+} = require("./dynamicTypes/elementsMap");
+const {
+  getWidgetsTypeDeclarations,
+  getWidgetDeclaration
+} = require("./dynamicTypes/widget");
```

Comment:
can we make all dynamicTypes expose the same api (getTypeDeclaration/getWhateverDeclaration) so that we can just import all via some index and map through them with the same function names ?

Reply:
for me, it would be much more understandable and organized.
then you can just import the full object and call the relevant function